pkg/github/github.go

```diff
@@ -93,15 +117,111 @@ func NewClient(ctx context.Context, token string, opts ...Option) *GitHubClient
 	return g
 }
 
-// CreateIssueComment creates a comment for an issue or pull request.
-func (g *GitHubClient) CreateIssueComment(ctx context.Context, owner, repo string, number int, body string) (*github.IssueComment, error) {
-	backoff := retry.NewFibonacci(g.cfg.initialRetryDelay)
-	backoff = retry.WithMaxRetries(g.cfg.maxRetries, backoff)
-	backoff = retry.WithCappedDuration(g.cfg.maxRetryDelay, backoff)
+// ListIssues lists all issues and returns their numbers in a repository matching the given criteria.
+func (g *GitHubClient) ListIssues(ctx context.Context, owner, repo string, opts *github.IssueListByRepoOptions) ([]*Issue, error) {
+	var uniqueResponses map[int]*Issue
+	var page *int // Use nil to indicate start of request.
+
+	pageStart := func(i *int) bool { return i == nil }
+	pageEnd := func(i *int) bool { return *i == 0 }
+	opt := *opts
 
-	var response *github.IssueComment
+	for pageStart(page) || !pageEnd(page) {
+		if err := g.withRetries(ctx, func(ctx context.Context) error {
+			if page != nil {
+				opt.Page = *page
+			}
+			issues, resp, err := g.client.Issues.ListByRepo(ctx, owner, repo, &opt)
+			if err != nil {
+				if _, ok := ignoredStatusCodes[resp.StatusCode]; !ok {
+					return retry.RetryableError(err)
+				}
+
+				return fmt.Errorf("failed to list issues: %w", err)
+			}
 
-	if err := retry.Do(ctx, backoff, func(ctx context.Context) error {
+			for _, i := range issues {
+				uniqueResponses[*i.Number] = &Issue{Number: *i.Number}
+			}
+			page = &resp.NextPage
+			return nil
+		}); err != nil {
+			return nil, fmt.Errorf("failed to list issues after retries: %w", err)
+		}
+	}
+
+	response := make([]*Issue, len(uniqueResponses))
+	i := 0
+	for _, r := range uniqueResponses {
+		response[i] = r
+		i++
+	}
```

Comment:
It would be more idiomatic to do:

```suggestion
	response := make([]*Issue, 0, len(uniqueResponses))
	for _, r := range uniqueResponses {
		responses = append(responses, r)
	}
```

Reply:
Done. This is what I wanted but the linter complained without the `0` arg.